### PR TITLE
test(policy): introduce new resources builders for tests

### PR DESCRIPTION
**Pull request by Automaat**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshretry/plugin/v1alpha1/plugin_test.go [hunk 1]
```diff
@@ -85,12 +77,19 @@ var _ = Describe("MeshRetry", func() {
...
 				},
+			}).WithPolicies(xds.MatchedPolicies{
+			Dynamic: map[core_model.ResourceType]xds.TypedMatchingPolicies{
+				api.MeshRetryType: {
+					Type:    api.MeshRetryType,
+					ToRules: given.toRules,
+				},
 			},
-		}
+		}).
```
Comment (bartsmykla) [inline pkg/plugins/policies/meshretry/plugin/v1alpha1/plugin_test.go:87]: Nit: I would pult `WithPolicies(...`

File: pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin_test.go [hunk 2]
```diff
@@ -459,17 +451,24 @@ var _ = Describe("MeshTimeout", func() {
...
+				api.MeshTimeoutType: {
+					Type:    api.MeshTimeoutType,
+					ToRules: given.toRules,
+				},
 			},
-			RuntimeExtensions: map[string]interface{}{},
-		}
-		Expect(gateway_plugin.NewPlugin().(core_plugins.ProxyPlugin).Apply(context.Background(), xdsCtx.Mesh, &proxy)).To(Succeed())
+		}).
+			Build()
```
Comment (bartsmykla) [inline pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin_test.go:462]: Nit: I would put `WithPolicies(...` to new line

**Review by bartsmykla — APPROVED** [review]

File: pkg/plugins/policies/meshloadbalancingstrategy/plugin/v1alpha1/plugin_test.go [hunk 3]
```diff
@@ -1493,6 +1421,24 @@ func createEndpointWith(zone string, ip string, extraTags map[string]string) cor
...
+			},
+			"payment": {
+				{
+					Tags: map[string]string{mesh_proto.ProtocolTag: core_mesh.ProtocolHTTP},
+				},
+			},
+		},
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshloadbalancingstrategy/plugin/v1alpha1/plugin_test.go:1441]: I will move it in next pr, that I have almost ready, I don't want to make this one too big

File: pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin_test.go (hunk 2 above)
Comment (Automaat) [reply pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin_test.go]: done

File: pkg/plugins/policies/meshretry/plugin/v1alpha1/plugin_test.go (hunk 1 above)
Comment (Automaat) [reply pkg/plugins/policies/meshretry/plugin/v1alpha1/plugin_test.go]: done

**Review by bartsmykla — APPROVED** [review]

**Pull request merged**